### PR TITLE
Always return variable in chainer.functions.noise.dropout

### PR DESCRIPTION
**Pull request by naoto0804**
This is a PR to wrap the result of dropout in test mode with `Variable`.

The reason is as follows.
In general, all functions in `chainer.functions` should return `Variable`. (which is also discussed in #3191)
However, in `chainer.functions.noise.dropout`, @ronekko kindly reported unexpected outputs like below in #3116:

```python
import numpy as np
import chainer

x = numpy.ones(5, 'f')
with chainer.using_config('train', False):
    y = chainer.functions.dropout(x)
    print(type(y))  # <class 'numpy.ndarray'>
    y = chainer.functions.dropout(chainer.Variable(x))
    print(type(y))  # <class 'chainer.variable.Variable'>
```
This is because input is returned without any modification in test mode in dropout.

### CODE REVIEW

**Comment by beam2d** [comment]
Could you use `as_variable` of #3218 instead of `identity`?

**Comment by naoto0804** [comment]
It seems that one test not relavant to this PR is failing in appveyor.

```
======================================================================
FAIL: test_double_backward_cpu (test_relu.TestReLU_param_0)  parameter: {'dtype': <type 'numpy.float16'>, 'shape': (3, 2)}
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Miniconda\lib\site-packages\chainer-3.0.0b1-py2.7.egg\chainer\testing\condition.py", line 74, in wrapper
    fail()
  File "C:\Miniconda\lib\site-packages\chainer-3.0.0b1-py2.7.egg\chainer\testing\condition.py", line 51, in fail
    instance.fail(msg)
AssertionError: 
Fail: 1, Success: 0
The first error message:
Traceback (most recent call last):
  File "C:\Miniconda\lib\site-packages\chainer-3.0.0b1-py2.7.egg\chainer\testing\condition.py", line 60, in <lambda>
    lambda: f(ins, *args[1:], **kwargs),
  File "C:\projects\chainer\tests\chainer_tests\functions_tests\activation_tests\test_relu.py", line 100, in test_double_backward_cpu
    self.check_double_backward(self.x, self.gy, self.ggx)
  File "C:\projects\chainer\tests\chainer_tests\functions_tests\activation_tests\test_relu.py", line 96, in check_double_backward
    **self.check_backward_options)
  File "C:\Miniconda\lib\site-packages\chainer-3.0.0b1-py2.7.egg\chainer\gradient_check.py", line 387, in check_double_backward
    dtype=dtype)
  File "C:\Miniconda\lib\site-packages\chainer-3.0.0b1-py2.7.egg\chainer\gradient_check.py", line 328, in check_backward
    testing.assert_allclose(gx, gx_accum, atol=atol, rtol=rtol)
  File "C:\Miniconda\lib\site-packages\chainer-3.0.0b1-py2.7.egg\chainer\testing\array.py", line 25, in assert_allclose
    x, y, atol=atol, rtol=rtol, verbose=verbose)
  File "C:\Miniconda\lib\site-packages\numpy\testing\utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "C:\Miniconda\lib\site-packages\numpy\testing\utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=0.001, atol=0.0001
(mismatch 100.0%)
 x: array(0.07129422575233171)
 y: array(0.07166847586631775)
-------------------- >> begin captured stdout << ---------------------
assert_allclose failed: 
  shape: () ()
  dtype: float64 float64
  i: (0,)
  x[i]: 0.0712942257523
  y[i]: 0.0716684758663
  err[i]: 0.000374250113986
x: 0.07129422575233171
y: 0.07166847586631775
--------------------- >> end captured stdout << ----------------------
----------------------------------------------------------------------
Ran 9852 tests in 62.142s
FAILED (SKIP=97, failures=1)
Command exited with code 1
```

**Comment by niboshi** [comment]
jenkins, test this please

File: chainer/functions/noise/dropout.py [hunk 1]
```diff
@@ -81,4 +82,4 @@ def dropout(x, ratio=.5, **kwargs):
 
     if configuration.config.train:
         return Dropout(ratio)(x)
-    return x
+    return as_variable(x)
```
Comment (okuta) [inline chainer/functions/noise/dropout.py]: Please use `return chainer.as_variable(x)`.

Comment (naoto0804) [reply chainer/functions/noise/dropout.py]: Reverted. Thank you!

**Comment by naoto0804** [comment]
Fixed. In appveyor, it seems that one test not relavant to this PR is failing.

**Comment by okuta** [comment]
Thank you!

**Pull request merged**